qiskit_experiments/visualization/plotters/curve_plotter.py

```diff
@@ -135,6 +138,42 @@ def _plot_figure(self):
                     )
 
             # Fit report
-            if "report_text" in self.supplementary_data:
-                report_text = self.supplementary_data["report_text"]
-                self.drawer.textbox(report_text)
+            report = self._write_report()
+            if len(report) > 0:
+                self.drawer.textbox(report)
+
+    def _write_report(self) -> str:
```

Comment:
I like the implementation of this method, especially that it allows subclasses to override or extend this functionality.